theano/gpuarray/opt.py

```diff
@@ -883,9 +883,8 @@ def local_gpua_softmaxwithbias(node, context_name):
 def local_assert(node, context_name):
 
     # Check if input nodes are already on the GPU
-    for i in node.inputs:
-        if isinstance(i.type, GpuArrayType):
-            return
+    if isinstance(node.inputs[0].type, GpuArrayType):
+        return
     return [host_from_gpu(node.op(as_gpuarray_variable(node.inputs[0],
```

Comment:
I tried doing that earlier and the test passed. I will add it here itself ? Because after this merge, that branch requires a rebase